safekeeper/src/timeline_manager.rs

```diff
@@ -579,18 +602,36 @@ impl Manager {
         }
     }
 
+    /// Reset partial backup state and remove its remote storage data. Since it
+    /// might concurrently uploading something, cancel the task first.
+    async fn backup_partial_reset(&mut self) -> anyhow::Result<Vec<String>> {
+        // Unevict timeline if it is evicted before erasing partial backup
+        // state.
+        self.unevict_timeline().await?;
```

Comment:
Simply calling `unevict_timeline()` is not what we want, otherwise we could just call `/digest` to call `wal_residence_guard()` which unevicts the timeline :)

We have a corrupted state in the remote storage, and uneviction will likely just fail in many cases.

What we need to do here is just this part of `do_uneviction()`:
```
    mgr.tli.switch_to_present().await?;
    // switch manager state as soon as possible
    mgr.is_offloaded = false;

    Ok(())
```
I'd called these functions directly, or made a separate function for this alongside `do_uneviction()`.

It works because we always store partial segments on disk (I haven't enabled setting with removing them anywhere), and we don't need to download anything.

Reply:
:facepalm: thanks for noticing, fixing.

Reply:
fixed